tools/clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -204,17 +204,16 @@ class ScalarExprEmitter
     if (BinaryOperator *BinOp = dyn_cast<BinaryOperator>(E)) {
       if (hlsl::IsHLSLMatType(E->getType()) ||
           hlsl::IsHLSLMatType(BinOp->getLHS()->getType())) {
-          if (BinOp->getOpcode() != BO_Assign) {
-              llvm::Value *LHS = CGF.EmitScalarExpr(BinOp->getLHS());
-              llvm::Value *RHS = CGF.EmitScalarExpr(BinOp->getRHS());
-              return CGF.CGM.getHLSLRuntime().EmitHLSLMatrixOperationCall(
-                  CGF, E, ConvertType(E->getType()), { LHS, RHS });
-          }
-          else {
+          if (BinOp->getOpcode() == BO_Assign) {
               LValue LHS = CGF.EmitLValue(BinOp->getLHS());
               llvm::Value *RHS = CGF.EmitScalarExpr(BinOp->getRHS());
               CGF.CGM.getHLSLRuntime().EmitHLSLMatrixStore(CGF, RHS, LHS.getAddress(), BinOp->getLHS()->getType());
               return RHS;
```

Comment:
Moving the handling of the assign operator just made the logic easier to follow since I'm adding another special case depending on the opcode